### ISSUE TITLE
[Bug] The item "obj\Debug\GitVersionInformation.g.cs" was specified more than once in the "Sources" parameter

### ISSUE THREAD

**Issue by riksking** [GitTools/GitVersion#2228]
Hello.

**Describe the bug**
I have C# project on .Net Framework 4.7.2.
I upgrade nuget from
`<package id="GitVersionTask" version="5.0.1" targetFramework="net472" developmentDependency="true" />`
to
`<package id="GitVersionTask" version="5.2.4" targetFramework="net472" developmentDependency="true" />`

When build I got this error:
> The item "obj\Debug\GitVersionInformation.g.cs" was specified more than once in the "Sources" parameter.  Duplicate items are not supported by the "Sources" parameter.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Current\Bin\Roslyn\Microsoft.CSharp.Core.targets	59	

I have another 33 projects (.Net Framework and .Net Core and .Net Standard) in this solution and all other projects build OK.

Any suggesions what's happen and how fix this?

## My Environment
Operating System Windows 10
Visual Studio 16.5.3

**Comment by prantzos** [comment]
I am having the same issue as well. I tried to figure out what is different about this one project and the "CoreCompile" task fails with this error.  Maybe resource files are to blame. On line 119 in "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\Roslyn\Microsoft.CSharp.Core.targets" the "Sources" parameter take a "Compile" parameter, and the "Csc" task (which fails too) has a condition "_CoreCompileResourceInputs.WithCulture != true".. I removed the resources and the condition, but I got the same error..

**Comment by marcusl** [comment]
FWIW, I get the same on the ServoPidControl project in my repo here: https://github.com/marcusl/ArduinoServoPID/commit/a169142cb617d64634213fab6c27f20bf0d76dcb

**Comment by arturcic** [comment]
@riksking I have tried to reproduce this issue like this:
I created a brand new console project targeting .net framework 4.7.2, added GitVersionTask version 5.0.1 as you described to packages.config (the old way of referencing nuget packages).

I performed several times build and in teh obj\Debug folder I saw several AssemblyInfo{random}.cs files basically on every build it was generating one. Also there was GitVersionInformation{random}.cs

The build was successful

Then I upgraded to latest 5.3.2 verison and it was still building, but now there is only one AssemblyInfo.g.s and one GitVersionInformation.g.s file that keeps changing if there are more commits. The old files were there (with {random}) but build was still succesful.

I then remove obj and bin folder and tried again to build, and still no issues to build. 

As far as I know we recently changed the way AssemblyInfo and GitVersionInformation is created, no random names so I would suggest cleaning up obj and bin folder and see if this helps. There was no way for me to repro this issue. If you can provide more details we can see what is the issue, for now closing the issue as I was not able to reproduce.

**Comment by riksking** [comment]
That's what I see in "obj\Debug"

![CommonUI_obj](https://user-images.githubusercontent.com/13674964/81560284-c275c980-9399-11ea-80af-2d14be6c1ff2.png)

I'm remove "bin" and "obj" and rebuild project several times, but got same error.

**Can you tell me what information to provide for error localization?**

Some details about my projects:
 I have two project (only these two relate to the problem in my opinion) "Common" and "Common.UI". 
Both in .net Framework 4.7.2, and Class Library type

![image](https://user-images.githubusercontent.com/13674964/81561927-a9224c80-939c-11ea-8d01-e300fa81d31c.png)


"Common.UI" have ProjectReference to "Common". And both have GitversionTask nuget package.

Also, I have AssemblyInfo.cs in each project and SharedAssemblyInfo.cs reference as a link

`<Link>Properties\SharedAssemblyInfo.cs</Link>`

AssemblyInfo.cs
`[assembly: AssemblyTitle("Common.UI")]`

SharedAssemblyInfo.cs
```
[assembly: AssemblyCompany("asdasdasd")]
[assembly: AssemblyProduct("asdasd")]
[assembly: AssemblyCopyright("Copyright ")]
[assembly: ComVisible(false)]
```

"Common" builds fine, but "Common.UI" build with error.

P.S. I try reproduce in new solution with same structure, but everything went well.
I do not know what could lead to such an error, but it is((

**Comment by arturcic** [comment]
I tried myself to reproduce the same structure - 2 projects with AssemblyInfo containing only the AssemblyTitle, and each using a SharedAssemblyInfo that is included as link wi the same content as you provided.

Initially install version 5.0.1 - build ok, then 5.3.2 - build ok. So I can't reproduce unfortunatelly. SO I will not continue working on this bug as I don't have a way to reproduce.

**Comment by prantzos** [comment]
I can verify that version 5.2.4 and the latest 5.3.2 that I just tried today, still have the same issue. BUT I tried 5.0.1 that was mentioned here and in my case it works and does not produce this error. 
In my case too the project that is not building is a class library targeting 4.7.2

![image](https://user-images.githubusercontent.com/8318629/81665451-0243bc00-9441-11ea-98d0-fe14a92b243b.png)

It has dependencies on WPF though which all the other projects that DO successfully build do not. I believe in both @riksking and in my case the UI projects have the issue and not the rest. Maybe this is a clue that can help @arturcic ? In your test you created an empty project. All of the library projects work just fine in our case as well, except when the library has WPF dependencies and in my case some resource files too. 

I will be more than glad to answer question so you can pinpoint the problem!

PS: In all my tests I delete bin and obj folders from the whole solution.

**Comment by arturcic** [comment]
@prantzos do you think you can create a sample project/repo with the issue you described having a dependency on WPF and share it with us?

**Comment by prantzos** [comment]
Sure @arturcic  I will create a new sample project with the same dependencies as the one that is not building and will share it.

**Comment by prantzos** [comment]
I copied 2 projects from the solution, removing all code but keeping as many dependencies possible. Sadly by doing so using both 5.0.1 and 5.3.2 the solution builds successfully. Which means I need to spend some more time to pinpoint where this breaks. 

Something that I did notice when updating from 5.0.1 to a newer 5.3.2, is that I get this error:

`
The result "" of evaluating the value "$(GitVersionAssemblyFile)" of the "AssemblyFile" attribute in element <UsingTask> is not valid.  C:\My\Workspaces\zivver\zivver-officeplugin\packages\GitVersionTask.5.0.1\build\GitVersionTask.targets
`

"GitVersionAssemblyFile" seems to be set as an empty string, which appears that none of the conditions were met in GitVersionTask.props, which is weird..

`
Condition="$(GitVersionAssemblyFile) == '' And '$(MSBuildRuntimeType)' != 'Core'"
Condition="$(GitVersionAssemblyFile) == '' And '$(MSBuildRuntimeType)' == 'Core'"
`

**Comment by marcusl** [comment]
Works for me now (5.1.3 -> 5.3.3) 👍 😀. Had to do a rebuild otherwise I got the same error.

**Comment by prantzos** [comment]
That is encouraging Marcus, and thank you for commenting about it. I usually rebuild though and while testing this issue I was always cleaning and then rebuilding and one time I even did a git clean -dfx to make sure there is nothing there.

**Comment by marcusl** [comment]
@prantzos I spoke too soon. I missed the fact that I needed to restart VS to finalize the update, then I got the same thing..  :-(

**Comment by prantzos** [comment]
Just saw that you still get the error which is the same for me and 5.3.3. I do not get it in 5.0.1 though and until I know what it is I am going to use 5.0.1.

**Comment by jbaehr** [comment]
During the build of a WPF Project, a temporary project (name "{original-project}_{some-random}_wpftmp.csproj", created by "GenerateTemporaryTargetAssembly") is generated which looks like a copy of the original project *with some additonal compile items at the end* (the content of "@(_GeneratedCodeFiles)"). Here, the GitVersionInformation.g.cs is already present. During the build of this temporary project (target "_CompileTemporaryAssembly"), the GitVersionTask is executed and adds the file another time to the compile items.

See https://referencesource.microsoft.com/#PresentationBuildTasks/BuildTasks/Microsoft/Build/Tasks/Windows/GenerateTemporaryTargetAssembly.cs,c0d174fe8373e3a0

I'll prepare an PR that somehow detect this situation and prevent the execution of "GenerateGitVersionInformation".

**Comment by jbaehr** [comment]
Here you go: #2280. To try it out you just need a few additional lines on the props-file of the nuget, i.e. no recompile.

**Comment by prantzos** [comment]
@jbaehr Thank you so much for investigating this! I understand your explanation about the duplicate GitVersionInformation file being generated and I verified it. 
I tested the solution you proposed in your PR and it seems to work!! I performed a clean rebuild and without the change in the props it failed. After applying the change and doing another clean build, I can verify it succeeded and the assembly file is versioned! 

I have one question though, so I can fully understand how this was playing out before the fix. 
In _GenerateTemporaryTargetAssembly.cs_ I can see that the generated code files are added. But I cannot find the target _"_CompileTemporaryAssembly"_ and _"@(_GeneratedCodeFiles)"_ in _"Microsoft.CSharp.Core.targets"_. Is this target on another file? 

Again, thank you so much for your help!!
